### PR TITLE
FEATURE: SD-390 Backward Compatibility for Combo chart

### PR DESCRIPTION
**Pull request by nadtan**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards)
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- https://github.com/gooddata/gdc-app-components/pull/537
- https://github.com/gooddata/gdc-analytical-designer/pull/2167

# Related Jira tasks
SD-390: https://jira.intgdc.com/browse/SD-390

### CODE REVIEW

**Comment by nadtan** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3126/ : FAILURE in 18m 51s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3872/ : SUCCESS in 11s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1663/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1025/ : UNSTABLE in 33m 45s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3873/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3127/ : FAILURE in 4m 56s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3875/ : SUCCESS in 13s

**Comment by nadtan** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3129/ : SUCCESS in 5m 33s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3877/ : SUCCESS in 16s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1665/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1028/ : UNSTABLE in 18m 42s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3878/ : SUCCESS in 17s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3130/ : SUCCESS in 9m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3879/ : SUCCESS in 11s

**Comment by nadtan** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1666/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1029/ : SUCCESS in 14m 03s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3880/ : SUCCESS in 12s

**Comment by imnutz** [comment]
LGTM

**Comment by nadtan** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-902.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1037/ : SUCCESS in 30m 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3899/ : SUCCESS in 52s

File: src/components/ComboChart.tsx [hunk 1]
```diff
@@ -44,11 +59,11 @@ export function ComboChart(props: IComboChartProps): JSX.Element {
...
         {
             localIdentifier: VIEW,
-            items: props.viewBy ? [props.viewBy] : [],
+            items: categories,
         },
     ];
 
-    const newProps = omit<IComboChartProps, IComboChartNonBucketProps>(props, [
+    const newProps = omit<IComboChartProps, IComboChartNonBucketProps>(clonedProps, [
         "primaryMeasures",
```
Comment (ivan-nejezchleb) [inline src/components/ComboChart.tsx]: missing removing "old" props here

File: stories/core_components/ComboChart.tsx [hunk 2]
```diff
@@ -127,6 +128,43 @@ storiesOf("Core components/ComboChart", module)
             </div>,
         ),
     )
+    .add("should override primaryMeasures & secondaryMeasures", () =>
+        screenshotWrap(
+            <ScreenshotReadyWrapper resolver={createHighChartResolver(3)}>
+                <div style={wrapperStyle} className="screenshot-container">
+                    <ComboChart
+                        projectId="storybook"
+                        columnMeasures={primaryMeasure}
```
[the remark this answers is not in the document]
Comment (nadtan) [reply stories/core_components/ComboChart.tsx]: Good point. Thank you

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by nadtan** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3152/ : SUCCESS in 10m 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3905/ : SUCCESS in 16s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1677/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1040/ : UNSTABLE in 15m 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3906/ : SUCCESS in 19s

**Comment by nadtan** [comment]
extended test - storybook

File: src/components/ComboChart.tsx [hunk 3]
```diff
@@ -30,8 +35,18 @@ type IComboChartNonBucketProps = Subtract<IComboChartProps, IComboChartBucketPro
...
  */
 export function ComboChart(props: IComboChartProps): JSX.Element {
-    const primaryMeasures = props.primaryMeasures || [];
-    const secondaryMeasures = props.secondaryMeasures || [];
+    const clonedProps = cloneDeep(props);
+    const { columnMeasures, lineMeasures, viewBy } = clonedProps;
+    const isOldConfig = Boolean(columnMeasures || lineMeasures);
+    const categories = isArray(viewBy) ? [viewBy[0]] : [viewBy];
+
+    if (isOldConfig) {
```
Comment (ivan-nejezchleb) [inline src/components/ComboChart.tsx]: can we check this logic by basic shallow enzyme test?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3153/ : SUCCESS in 9m 53s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3907/ : SUCCESS in 15s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1678/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1041/ : SUCCESS in 21m 58s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3908/ : SUCCESS in 14s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3156/ : SUCCESS in 6m 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3910/ : SUCCESS in 10s

File: src/components/ComboChart.tsx (hunk 3 above)
Comment (nadtan) [reply src/components/ComboChart.tsx]: Yes, please help to check again

File: src/components/ComboChart.tsx (hunk 1 above)
Comment (nadtan) [reply src/components/ComboChart.tsx]: updated

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3157/ : FAILURE in 48m 33s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3911/ : SUCCESS in 29s

File: src/components/tests/ComboChart.spec.tsx [hunk 4]
```diff
@@ -19,4 +20,49 @@ describe("ComboChart", () => {
         );
         expect(wrapper.find(AfmComboChart)).toHaveLength(1);
     });
+
+    it("should render with columnMeasures & lineMeasures", () => {
+        const wrapper = shallow(<ComboChart projectId="foo" columnMeasures={[M3]} lineMeasures={[M4]} />);
+        expect(wrapper.find(AfmComboChart)).toHaveLength(1);
```
Comment (ivan-nejezchleb) [inline src/components/tests/ComboChart.spec.tsx]: I would expect, that this will test 'props' of AfmComboChart, not only its presence in react tree

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3163/ : SUCCESS in 3m 29s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3914/ : SUCCESS in 10s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3179/ : SUCCESS in 3m 01s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3930/ : SUCCESS in 11s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1696/ : UNSTABLE in 13m 34s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3180/ : SUCCESS in 2m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/873/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/883/ : SUCCESS in 21m 22s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1697/ : SUCCESS in 16m 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3181/ : SUCCESS in 2m 56s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/874/ : SUCCESS in 22s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/884/ : FAILURE in 12m 32s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1698/ : UNSTABLE in 14m 30s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3182/ : SUCCESS in 3m 32s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/875/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/885/ : SUCCESS in 21m 00s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1699/ : SUCCESS in 14m 30s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3184/ : SUCCESS in 7m 39s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/876/ : SUCCESS in 30s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/886/ : SUCCESS in 15m 20s

**Pull request merged**